src/UnsupportedFeatureException.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace React\EventLoop;
+
+use Exception;
+
+final class UnsupportedFeatureException extends Exception
+{
+    public static function create($feature, $loopName)
+    {
+        return new self('Event loop feature "' . $feature . '" isn\'t supported by the "' . $loopName . '" event loop');
+    }
+}
```

Comment:
This adds the first custom exception to this package, so I would much rather discuss this in a separate PR and remove here for now? :+1: A `BadMethodCallException` seems to be a good replacement for now.

Reply:
Why Bad method call?

Reply:
@kelunik What was your suggestion again?

Reply:
> Exception thrown if a callback refers to an undefined method or if some arguments are missing.

`BadMethodCallException` is usually used in combination with `__call`, not if requirements for a call are missing.

My suggestion is to keep a custom exception that makes things clear.

Reply:
@clue personally I moving more and more towards custom exceptions for each use case so the error handled in a specific manner for that error. Maybe we should have a discussion about if we want that in ReactPHP before adding a custom exception to a packages. I'll change it unless you and @jsor prefer this idea of a custom exception per error.

Reply:
I'm not suggesting that "custom exception per error" is not a good idea (see above), merely that I think it makes sense to discuss this in a separate ticket because I would love to get this feature in already :shipit: :-) Also note that this is the first method can may throw in the first place.

Reply:
Alright I've swap it out with a core exception and we can have this discussion in another issue so we can get this feature in :shipit: